docs/preflight/v1beta3-overview.md

```diff
@@ -0,0 +1,505 @@
+kind: Preflight
+metadata:
+  name: basic-preflight
+spec:
+  collectors:
+    - clusterResources: {}
+
+  analyzers:
+    {{- if .Values.kubernetes.enabled }}
+    - docString: |
+        Title: Kubernetes Version
+        Requirement:
+          - Minimum: {{ .Values.kubernetes.minVersion }}
+        Ensures the cluster meets minimum API requirements.
+      clusterVersion:
+        checkName: Kubernetes version
+        outcomes:
+          - fail:
+              when: '< {{ .Values.kubernetes.minVersion }}'
+              message: Requires Kubernetes {{ .Values.kubernetes.minVersion }} or later
+          - pass:
+              message: Kubernetes version meets requirements
+    {{- end }}
+
+    {{- if .Values.storage.enabled }}
+    - docString: |
+        Title: Default Storage Class
+        Requirement:
+          - StorageClass "{{ .Values.storage.className }}" must exist
+        Enables dynamic volume provisioning for application data.
+      storageClass:
+        checkName: Default StorageClass
+        storageClassName: '{{ .Values.storage.className }}'
+        outcomes:
+          - fail:
+              message: StorageClass not found
+          - pass:
+              message: StorageClass {{ .Values.storage.className }} is available
+    {{- end }}
+
+    {{- if .Values.nodes.enabled }}
+    - docString: |
+        Title: Minimum Node Count
+        Requirement:
+          - At least {{ .Values.nodes.minimum }} nodes required
+        Ensures sufficient capacity for high availability.
+      nodeResources:
+        checkName: Node count
+        outcomes:
+          - fail:
+              when: 'count() < {{ .Values.nodes.minimum }}'
+              message: Requires at least {{ .Values.nodes.minimum }} nodes (found {{ "{{" }} count() {{ "}}" }})
+          - pass:
+              message: Cluster has sufficient nodes
+    {{- end }}
+```
+
+**Corresponding values file (values.yaml):**
+
+```yaml
+kubernetes:
+  enabled: true
+  minVersion: "1.24.0"
+
+storage:
+  enabled: true
+  className: "standard"
+
+nodes:
+  enabled: true
+  minimum: 3
+```
+
+**Run it:**
+
+```bash
+preflight basic-preflight.yaml --values values.yaml
+```
+
+## Advanced Example: Dynamic Messages
+
+This example shows how to create **dynamic docStrings and messages** that adapt based on actual cluster state.
+
+Imagine checking for sufficient memory across nodes, but providing specific feedback about what's needed:
+
+```yaml
+apiVersion: troubleshoot.sh/v1beta3
+kind: Preflight
+metadata:
+  name: advanced-preflight
+spec:
+  collectors:
+    - clusterResources: {}
+
+  analyzers:
+    {{- if .Values.resources.memory.enabled }}
+    - docString: |
+        Title: Node Memory Requirements
+        Requirement:
+          - Each node must have at least {{ .Values.resources.memory.minPerNodeGi }} GiB memory
+          - Total cluster memory must be at least {{ .Values.resources.memory.totalMinGi }} GiB
+
+        Rationale:
```

Comment:
In the netlify preview this renders a little strange. Makes it looks like the yaml is formatted strangely. not sure if there's anything you can do about that or not.